packages/ngtools/webpack/src/ngcc_processor.ts

```diff
@@ -57,13 +60,15 @@ export class NgccProcessor {
 
       return;
     }
-    const normalizedJsonPath = workaroundResolve(packageJsonPath);
 
     const timeLabel = `NgccProcessor.processModule.ngcc.process+${moduleName}`;
     time(timeLabel);
     this.ngcc.process({
-      basePath: normalizedJsonPath.substring(0, normalizedJsonPath.indexOf(moduleName)),
-      targetEntryPointPath: moduleName,
+      basePath: this._nodeModulesDirectory,
+      targetEntryPointPath: path.relative(
+        this._nodeModulesDirectory,
+        path.dirname(packageJsonPath),
+      ),
```

Comment:
Inside ngcc we actually resolve `targetEntryPointPath` against `basePath`, and only then use that resolved path. So you could actually just pass the absolute path for `targetEntryPointPath`.

Reply:
In that case, the comment is misleading cause it states that it should be relative.

https://github.com/angular/angular/blob/b70d20b51085ab4f86751d4f8c1e92e76da9356e/packages/compiler-cli/ngcc/src/main.ts#L34

Reply:
Yes it is misleading. We could change that to something like:

> The path to the primary package to be processed. If not absolute then it must be relative to `basePath`.